### PR TITLE
PromQL: Parse grouping opts with the generated parser

### PR DESCRIPTION
**Pull request by slrtbtfs**
This PR extends the generated parser to parse grouping opts.

This turned out to be more complex than expected since the lexer will return keyword tokens for labels with names that are also keywords.

To work around this, an additional `maybe_label` grammar rule is introduced, that matches on any token that could be a label.

Based on this PR, implementing #6470 will be a two line change (excluding generated code).

This is part of the parser rewrite effort described in #6410 .

cc @brian-brazil @simonpasquier 
Signed-off-by: Tobias Guggenmos <tguggenm@redhat.com>

### CODE REVIEW

File: promql/lex.go [hunk 1]
```diff
@@ -123,6 +123,10 @@ func (i ItemType) isRightAssociative() bool {
 
 type ItemType int
 
+// This is a list of all keywords in PromQL
+// When changing this list, make sure to also change
+// the maybe_label grammar rule in the generated parser
+// to avoid misinterpretation of labels as keywords
```
Comment (brian-brazil) [inline promql/lex.go]: Full stop.

Comment (slrtbtfs) [reply promql/lex.go]: Fixed it.

Given how often this happens to me, I maybe should write some script to check for this specific error automatically.

**Comment by slrtbtfs** [comment]
> I take it you're leaving the commas for a future PR?

Yes, as an immediate follow up.

Thought it would be nicer to have the implementation and behavior changes in separate PRs.

**Comment by brian-brazil** [comment]
:+1: 

CI seems a bit borked.

**Comment by brian-brazil** [comment]
Can you try making a trivial change and force pushing?

**Comment by slrtbtfs** [comment]
Adding an empty commit to trigger the CI worked.

**Comment by brian-brazil** [comment]
Thanks!

**Pull request merged**